### ISSUE TITLE
CellML API: inconsistent code generation

### ISSUE THREAD

**Issue by aram148** [opencor/opencor#1947]
The following errors has been found while investigating the saucerman 2003 model:

1. The removal of the model ID annotation from a working saucerman 2003 cellml file stops the model from working. If the model ID is then replaced, the model seems to work again.

2.  If the contents of the working cellml file is pasted into a new cellml file, the model cannot then be made to work.

3. If a copy of the cellml file is made outside of the opencor environment, the copied file runs and can be solved (issue 1 still holds in this case).

**Comment by agarny** [comment]
Ok, having looked at the [CellML API code](https://github.com/opencor/cellml-api), I now understand what is going on:
1. The model element of the [saucerman 2003 cellml file](https://github.com/opencor/opencor/files/2753846/new_saucerman_brunton_michailova_mcculloch_2003.txt) has a cmeta:id, and the CellML API uses that cmeta:id to retrieve annotation that contains hints for the initial values to be used for the model... and, indeed, the [saucerman 2003 cellml file](https://github.com/opencor/opencor/files/2753846/new_saucerman_brunton_michailova_mcculloch_2003.txt) contains such annotation! So, indeed, if we remove the cmeta:id for the model element, then the CellML API is going to use `0.1001` as an initial guess, which in the case of the present model is clearly not a good guess, hence it fails.
2. This also makes sense since by copying the contents of the CellML file using the CellML Text view, we are not copying the annotations, so even though the model element has a cmeta:id, the CellML API cannot find the annotation associated with it, hence the model cannot be solved.
3. This also makes sense since physically copying that file obviously preserves the annotation.

Bottom line: it all makes sense. I just didn't expect the CellML API to rely on annotation to get the initial values. In fact, I don't think the code generator should provide those initial guesses. It should be the solver's job to come up with those initial guesses. Oh well... closing this issue since we are anyway switching to libCellML... someday...
